gateway/resolver/subscription.go

```diff
@@ -192,15 +200,18 @@ func (r *Service) runWatch(
 					if name != "" {
 						singleObj = previousObjects[namespace+"/"+name]
 					}
+
+					var data interface{}
+					if singleObj == nil { // object will be nil in case it is deleted
+						data = nil
+					} else {
+						data = singleObj.Object
+					}
```

Comment:
Wouldn't this be easier to understand as it's `nil` anyway on initalization:
```suggestion
					if singleObj != nil { // object will be nil in case it is deleted
						data = singleObj.Object
					}
```